framework/src/bundle/BundleArchive.cpp

```diff
@@ -116,29 +137,63 @@ std::vector<BundleResource> BundleArchive::FindResources(
 
 BundleArchive::TimeStamp BundleArchive::GetLastModified() const
 {
-  return TimeStamp() + std::chrono::milliseconds(data->lastModified);
+  return TimeStamp { std::chrono::milliseconds(lastModified) };
 }
 
 void BundleArchive::SetLastModified(const TimeStamp& ts)
 {
-  data->lastModified =
-    std::chrono::duration_cast<std::chrono::milliseconds>(ts.time_since_epoch())
-      .count();
+  namespace sc = std::chrono;
+  
+  lastModified = sc::duration_cast<sc::milliseconds>(ts.time_since_epoch()).count();
 }
 
 int32_t BundleArchive::GetAutostartSetting() const
 {
-  return data->autostartSetting;
+  return autostartSetting;
 }
 
 void BundleArchive::SetAutostartSetting(int32_t setting)
 {
-  data->autostartSetting = setting;
+  autostartSetting = setting;
 }
 
 std::shared_ptr<BundleResourceContainer>
 BundleArchive::GetResourceContainer() const
 {
   return resourceContainer;
 }
+
+const AnyMap& BundleArchive::GetManifest() const
```

Comment:
Not sure what to name this. This is the unchanged name of the API, and it used to simply "return manifest.GetHeaders();". The code in here that reads/parses the manifest used to be in the constructor but was moved here because it's now only read from the file if it's not constructed with a manifest in the first place... so it acts just like it used to from the outside. If you're adamant about changing the name, do you have a suggestion what the new name should be?

Reply:
Renamed to GetInjectedManifest. Not sure if that's any better.